lib/BitcastUtils.cpp

```diff
@@ -1091,10 +1085,47 @@ GetIdxsForTyFromOffset(const DataLayout &DataLayout, IRBuilder<> &Builder,
     startIdx = 1;
   }
 
+  // TODO: Is there a better way to adjust for this?
+  if (DynVal == nullptr) {
+    if (TyBitWidths.size() == 1 &&
+        (TyBitWidths[0] > CstVal * SmallerBitWidths) && SrcTy->isStructTy()) {
+      // Descending into the aggregate so 0 the first GEP index.
+      if (startIdx == 0)
+        Idxs.push_back(Builder.getInt32(0));
+
+      auto *ty = SrcTy;
+      while (ty->isStructTy() &&
```

Comment:
I am not sure about this loop, I don't see any test testing it (amongst those you have added. Maybe one already existing does)

Reply:
I'll add an extra test case that covers this.